web_scraping/browser.py

```diff
@@ -0,0 +1,90 @@
+import os
+import json
+import time
+from concurrent.futures import ThreadPoolExecutor
+from selenium import webdriver
+
+
+def create_output_dir(directory="output"):
+    if not os.path.exists(directory):
+        os.mkdir(directory)
+
+
+def get_urls_from_file(input_file="input/urls.input"):
+    if not os.path.exists(input_file):
+        raise FileNotFoundError(f"{input_file} does not exist")
+
+    with open(input_file, "r") as file:
+        return file.read().splitlines()
+
+
+def initialize_driver():
+    return webdriver.Chrome()
+
+
+def get_html_content(driver):
+    return driver.page_source
+
+
+def get_resources(driver):
+    resources = set()
+    for resource in driver.execute_script("return window.performance.getEntriesByType('resource')"):
+        resources.add(resource['name'])
```

Comment:
Use list comprehension (or set comprehension in this case)
```suggestion
        {resource['name'] for resource in drive...}
```